r/R/dplyr.R

```diff
@@ -540,6 +542,31 @@ arrow_stringr_string_replace_function <- function(FUN, max_replacements) {
   }
 }
 
+arrow_r_string_split_function <- function(FUN, reverse = FALSE, max_splits = -1){
+  function(x, split, fixed = FALSE, perl = FALSE, useBytes = FALSE){
+    
+    assert_that(is.string(split))
+    
+    # if !fixed but no regex metachars in split pattern, allow to proceed as split isn't regex
+    if(!fixed && contains_regex(split)){
+      stop("Regular expression matching not supported in strsplit for Arrow", call. = FALSE)
+    }
+    if(fixed && perl){
+      warning("argument 'perl = TRUE' will be ignored")
```

Comment:
Let's use `call. = FALSE` in `warning()` too:
```suggestion
      warning("Argument 'perl = TRUE' will be ignored", call. = FALSE)
```